circuit_knitting/utils/equivalence.py

```diff
@@ -0,0 +1,155 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2024.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Equivalence utilities.
+
+.. currentmodule:: circuit_knitting.utils.equivalence
+
+.. autosummary::
+   :toctree: ../stubs/
+
+"""
+
+import numpy as np
+from qiskit.circuit import (
+    EquivalenceLibrary,
+    QuantumCircuit,
+    QuantumRegister,
+    Parameter,
+)
+from qiskit.circuit.library.standard_gates import (
+    RZGate,
+    XGate,
+    YGate,
+    ZGate,
+    HGate,
+    SGate,
+    IGate,
+    SdgGate,
+    SXGate,
+    SXdgGate,
+    TGate,
+    TdgGate,
+    RXGate,
+    RYGate,
+    PhaseGate,
+)
+
+_eagle_sel = HeronEquivalenceLibrary = EagleEquivalenceLibrary = EquivalenceLibrary()
+
+########## Single-qubit Eagle native gate set: x, sx, rz, i ##########
```

Comment:
I think it's better to delete the native gates from the equivalence library. Qiskit equivalence libraries don't return the gate itself, only alternate equivalences.